docs/ansible/roles/debops.postldap/guides.rst

```diff
@@ -28,7 +28,99 @@ email addresses. The default configuration uses first the aliases set by
 See also :ref:`debops.dovecot` and :ref:`debops.roundcube` for an IMAP server
 and Email-Webclient correspondingly.
 
+The following example shows a real-world™ setup on the Hetzner Cloud. It consists of two servers,
+one ``controller`` and a ``mail-server``. The LDAP directory is hosted also in the ``controller``.
+``mail-server`` has access to LDAP over an internal network (10.10.10.0/24) attached directly to the VMs.
+This setup has no internal DNS server (no split-DNS), thus internal IPs are mapped to DNS entries in the form ``$service.hetzner.mydomain.net``.
+See `secret__ldap_fqdn`.
+
+.. code-block:: yaml
+  :caption: ansible/inventory/group_vars/hetzner/ldap.yml
+
+  ---
+
+  # Enable LDAP, as is deactivated by default
+  ldap__enabled: True
+
+  ldap__domain: 'mydomain.net'
+
+  # Domain used for LDAP base DN and to select default LDAP server.
+  secret__ldap_domain: 'mydomain.net'
+
+  # LDAP server hostname / internal IP address which holds the database.
+  # ldap_* modules will connect to it natively, so it should be available at least from the Ansible Controller.
+  secret__ldap_fqdn: 'ldap.hetzner.mydomain.net'
+
+.. code-block:: yaml
+  :caption: ansible/inventory/host_vars/skynet.mydomain.net/slapd.yml
+
+  ---
+
+  ## Network access to OpenLDAP server
+  # Firewall Settings
+  #   Block connections to the OpenLDAP via system firewall and TCP Wrappers from any host (aka Internet);
+  #   Hosts that can connect must be specified via the slapd__*_allow variables.
+  slapd__accept_any: false
+
+  slapd__group_allow:
+    # IPv4/v6 localhost
+    - '127.0.0.1'
+    - '::1'
+    - '{{ ansible_default_ipv4.address
+          if (ansible_default_ipv4|d() and ansible_default_ipv4.address|d() )
+          else "127.0.0.1" }}'
+    - '{{ ansible_default_ipv6.address
+          if (ansible_default_ipv6|d() and ansible_default_ipv6.address|d() )
+          else "::1" }}'
```

Comment:
The above 4 entries shouldn't be needed. The `debops.ferm` firewall already accepts all traffic on `localhost` explicitly.

I'm not sure why would you want to talk to the `slapd` service over external IP addresses from the same host. But if you found out that this is needed for some reason, I think that a better solution would be to put the `ansible_default_ipv*` entries in the `debops.slapd` role, probably in a separate variable that gets activated when the public access to the OpenLDAP service is disabled. Just need a good use case for this.

Reply:
> I'm not sure why would you want to talk to the `slapd` service over external IP addresses from the same host. But if you found out that this is needed for some reason,

The ext IPs rules are needed to access the SLAPd over SSH-Port-Forwarding. So I am able to use ApacheDirectoryStudio to modify LDAP attributes.

> I think that a better solution would be to put the `ansible_default_ipv*` entries in the `debops.slapd` role, probably in a separate variable that gets activated when the public access to the OpenLDAP service is disabled. Just need a good use case for this.

See above. But I do not think it is a good reason to be a default. Instead write down this solution in a Knowledge Base.

@drybjed What do prefer: keep or remove ?

Reply:
Removed for the sake of simplicity, however the question with the SSH PF still remains

Reply:
Ah, SSH port forwarding... I see. I just connect to the OpenLDAP server directly, over TLS. You could open the firewall if you have static IP address. For dynamic IP address, Tinc VPN could be a good solution, so that you have access to the internal network.

Wtih SSH port forwarding, I think that since you forward the TCP port to `localhost`, the firewall should work fine without additional rules. If you show what `ssh` command you use to create the tunnel, or other needed configuration, I could test it locally to see if you really needed to open the host's external IP addresses.

Reply:
Yeah, tinc is in the todo list...